src/oidcmsg/time_util.py

```diff
@@ -200,7 +202,8 @@ def time_a_while_ago(
     :return: datetime instance using UTC time
     """
     delta = timedelta(days, seconds, microseconds, milliseconds, minutes, hours, weeks)
-    return datetime.utcnow() - delta
+    res = datetime.now(timezone.utc) - delta
+    return res.replace(tzinfo=None)
```

Comment:
why do we need to strip the timezone information?

Reply:
naive and aware datetime can't be compared, so for retrocompatibility I operate with aware datetime but always return naive datetime

Reply:
considering that standard requries utc datetime, it's not important to have it with timezone

Reply:
What are those comparisons?

It is important it to have it with timezone to ensure that operations on time are done in the needed timezone.
I would argue we need to do the opposite - have all datetimes in UTC to ensure calculations are correct. So, the comparisons that take place, should also have dates loaded as UTC.

Reply:
the comparison doesn't happen here but later in a unit test

````
______________________________________________________________________________________________ test_time_a_while_ago _______________________________________________________________________________________________

    def test_time_a_while_ago():
        dt = datetime.utcnow()
        t = time_a_while_ago(seconds=10)
>       delta = dt - t  # slightly less than 10
E       TypeError: can't subtract offset-naive and offset-aware datetimes

tests/test_03_time_util.py:204: TypeError
````

considering that standard require UTC timestamp I didn't see any needs to carrying aware datetimes in the code BUT you kow that I'ma good boy that can spend some more effort to get all happy :)

Roland approved, do we have to put another change or do you think that's good enough as is?

Reply:
If we are going to change `.utcnow()` to `.now(utc)` I think we should do it everywhere.
If we don't want to do it, then let's not do it at all.

Reply:
I don't know the conseguences to other libraries and behaviours 
I preferred to let the type be the same as before to prevent any breaking changes with this release

that's the answer, feel free to open an issue to keep track of this